src/figuresClasses.ts

```diff
@@ -1,11 +1,71 @@
-export interface Figure {}
+export interface Figure {
+  shape: string;
+  color: string;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape = 'triangle';
+
+  constructor(
+    public color: string,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Sides must be greater than 0');
+    }
+
+    if (a + b <= c || a + c <= b || b + c <= a) {
+      throw new Error('Triangle with such sides does not exist');
+    }
+  }
+
+  getArea(): number {
+    const p = (this.a + this.b + this.c) / 2;
+
+    return +Math.sqrt(p * (p - this.a) * (p - this.b) * (p - this.c)).toFixed(
+      2,
+    );
```

Comment:
I would format it something like that. It's not the best approach, you can probably find better. But we want to have this construction readable to understand exactly what's happening here

```suggestion
    return +Math.sqrt(
      p * (p - this.a) * (p - this.b) * (p - this.c)
    ) .toFixed(2);
```